lib/rdoc.rb

```diff
@@ -120,6 +120,29 @@ def self.load_yaml
     end
   end
 
+  def self.home
+    homedir = begin
+                File.expand_path('~')
+              rescue ArgumentError
+              end
+    homedir ||= ENV['HOME'] ||
+                ENV['USERPROFILE'] || ENV['HOMEPATH'] # for 1.8 compatibility
+    if homedir then
+      rdoc_dir = File.join homedir, ".rdoc"
+    end
+
+    if File.exist?(rdoc_dir)
+      rdoc_dir
+    else
+      # XDG
+      xdg_data_home = ENV["XDG_DATA_HOME"] || File.join(Gem.user_home, '.local', 'share')
```

Comment:
Should we use `homedir` here, for consistency on "home path" calculation, and so that we don't introduce an unnecessary dependency on rubygems?

Reply:
Ah, Right. I'll update it.